runtime/parser/lexer/token.go

```diff
@@ -23,22 +23,20 @@ import (
 )
 
 type Token struct {
-	Type  TokenType
-	Value any
+	Type         TokenType
+	SpaceOrError any
 	ast.Range
 }
 
 func (t Token) Is(ty TokenType) bool {
 	return t.Type == ty
 }
 
-func (t Token) IsString(ty TokenType, s string) bool {
-	if !t.Is(ty) {
-		return false
+func (t Token) Source(input []byte) []byte {
+	endOffset := t.EndPos.Offset + 1
+	inputLength := len(input)
+	if endOffset > inputLength {
+		endOffset = inputLength
```

Comment:
Good point! This was only necessary because tests where calling `Token.Source()` on all tokens, including EOF, which is "out-of-bounds" of the input (source code).

Refactored in [1ef8ff1](https://github.com/onflow/cadence/pull/1930/commits/1ef8ff1dba7bfa3721d1fb15509863e8b072c843)